core/domain/prod_validation_jobs_one_off.py

```diff
@@ -6183,6 +6184,83 @@ def _get_external_instance_custom_validation_functions(cls):
         return [cls._validate_reference]
 
 
+class PlatformParameterModelValidator(BaseModelValidator):
+    """Class for validating PlatformParameterModel."""
+
+    @classmethod
+    def _get_model_id_regex(cls, unused_item):
+        return r'^[A-Za-z0-9_]{1,100}$'
```

Comment:
Just to check -- are these parameters defined within the code, similar to how ConfigProperties are currently defined? If so, make sure to have a backend test that iterates through all such platform parameters and validates this. If not, make sure that this validation is done prior to when a user creates a new platform parameter.

Also, you will need a plan for ensuring that the IDs of existing properties don't change (either they should not be developer-editable or they should not be user-editable in the cases above, respectively) and the design doc should detail a plan for deprecating old properties. /cc @BenHenning

Reply:
Yes they are defined in the code, at the domain level, which is also similar to ConfigProperties. I think such guards should be implemented at the domain level (maybe raising an exception when the name doesn't match the regex), at the storage level we already have the audit jobs.
If the name of an existing parameter is changed, it can be seem as an unexpected deletion because we are no longer able to find the parameter using the original name, so the test against accidental deletion is applicable for these cases -- tests that try to use the deleted parameter will fail.